cnxarchive/views.py

```diff
@@ -981,3 +1008,46 @@ def robots(request):
     resp.expires = html_date(exp_time)
     resp.body = robots_dot_txt.to_string()
     return resp
+
+
+@view_config(route_name='recent', request_method='GET',
+             accept="application/json",
+             renderer='templates/recent.rss')
+def recent(request):
+    # setting the query variables
+    num_entries = request.GET.get('number', 10)
+    start_entry = request.GET.get('start', 0)
+    portal_type = request.GET.get('type', ['Collection', 'Module'])
+    if portal_type != ['Collection', 'Module']:
+        portal_type = [portal_type]
+    # search the database
+    settings = request.registry.settings
+    statement = """
+                SELECT name, revised, authors, abstract,
+                'http://cnx.org/contents/'||
+                ident_hash( uuid, major_version, minor_version) AS link
```

Comment:
It'd be better to use do this url building outside of the query statement. For instance, build the ident_hash in the query (i.e. what you've got without the url prefix concatenation), then on row retrieval use the framework to build the correct url (e.g. `rows = cursor.fetchall(); request.route_url(rows[0][4])`). This will make the url dynamic to whatever environment we put the code in.